### PR TITLE
Refactor leaderboard code

### LINKED ISSUE #948: Leaderboard test fails due to undownloadable artifact
This test was commented out in #946

### LINKED ISSUE #956: Replace leaderboard with ClimaAnalysis leaderboard
Most of the leaderboard code has been ported and improved in ClimaAnalysis 0.5.8. When that is released, it would be good to switch to using that.

### PR DESCRIPTION
**Pull request by ph-kev**
closes #948 , closes #956 

This PR refactors the leaderboard code using the new features from ClimaAnalysis. This commit also deletes the old leaderboard code and the tests for it. Also, there is an off by one month issue when handling the dates and seasons. This issue arises because the first day of each month in the simulation data represents the monthly average of the previous month. This issue only affect the plots involving seasonal data. This is fixed in this commit.

The commit also moves the code to leaderboard.jl and a line is added to the pipeline to run the script.

One significant difference is the leaderboard which now plot the best and worst single model using only annual rather than averaging the error over the annual and seasonal data.

The leaderboard now looks like this:
bias_leaderboard

To compare, the previous leaderboard looks like this:
bias_leaderboard

The errors for the seasonal data change slightly because of the off by one month issue. The new bias plot for MAM looks like this:
bias_pr_MAM

The old bias plot for MAM looks this:
bias_pr_MAM

In addition to the plots above, there is now a plot with all the seasons for a subset of the variables (see `compare_vars_biases_groups` for which variables are plotted together in data_sources.jl).
bias_pr_all_seasons

### CODE REVIEW

**Review by Sbozzolo — COMMENTED** [review]
Would it be possible to split the `leaderboard.jl` file in such a way that if one wants to add a new variable they would have to only change one file? (Essentially, moving the loaders to a separate file)

**Comment by ph-kev** [comment]
> Would it be possible to split the `leaderboard.jl` file in such a way that if one wants to add a new variable they would have to only change one file? (Essentially, moving the loaders to a separate file)

I moved all the code involving data and preprocessing to `data_sources.jl`.

File: experiments/ClimaEarth/leaderboard/data_sources.jl
Comment (Sbozzolo) [inline experiments/ClimaEarth/leaderboard/data_sources.jl:59]: Maybe I am getting confused, but shouldn't this be `shift_to_start_of_month`?

File: experiments/ClimaEarth/leaderboard/leaderboard.jl
Comment (Sbozzolo) [inline experiments/ClimaEarth/leaderboard/leaderboard.jl:27]: Instead of assuming that `pr` is available, you can pick the first entry in the dict

File: experiments/ClimaEarth/leaderboard/leaderboard.jl
Comment (Sbozzolo) [inline experiments/ClimaEarth/leaderboard/leaderboard.jl:101]:
```suggestion
                CairoMakie.save(
                    joinpath(leaderboard_base_path, "bias_$(first(compare_vars_biases))_$season.png"),
                    fig_bias,
                )
```
It's fine to change the name to _ANN

**Review by Sbozzolo — COMMENTED** [review]
I think this looks fantastic. Can you add an entry to the NEWS file to briefly mention to script? And can you add a section somewhere in the docs about "How do I add a new variable to the leaderboard?"

**Review by Sbozzolo — APPROVED** [review]
LGTM. Thank you! 

I'd recommend you tag Julia for a second review before merging.

File: experiments/ClimaEarth/run_amip.jl
Comment (juliasloan25) [inline experiments/ClimaEarth/run_amip.jl:860]: The comments in this section are really helpful! Do you think you could move them to the docs? We should add a new section `Model Output` (or something like that) and add a page about the leaderboard that contains this information.

The AMIP driver (run_amip.jl) is pretty complex right now but we're moving toward slimming it down, so it would be better to move a big comment like this to the docs

File: experiments/ClimaEarth/leaderboard/leaderboard.jl
Comment (juliasloan25) [inline experiments/ClimaEarth/leaderboard/leaderboard.jl:10]: these comments are duplicates of the docstring below

**Review by juliasloan25 — COMMENTED** [review]
This looks great, thank you Kevin! I just had a comment about moving some information to the docs, but after that it looks good to merge

**Review by juliasloan25 — APPROVED** [review]
Looks good! I'm not sure why the GPU slabplanet runs fail - is this rebased on the latest main?

**Comment by ph-kev** [comment]
> Looks good! I'm not sure why the GPU slabplanet runs fail - is this rebased on the latest main?

I don't think this is rebased on the latest main. I am going to rebase and push right now.

**Pull request merged**
